### ISSUE TITLE
postgres driver requires psycopg2

### ISSUE THREAD

**Issue by kning** [datafold/data-diff#180]
Getting the following error trying run a diff after installing the postgres driver

![image](https://user-images.githubusercontent.com/1820651/181296619-a8b5287e-7b83-4b6f-84f5-888f3446dff3.png)

Think it just needs to be added to requirements for the driver.

**Comment by erezsh** [comment]
Because we support a lot of databases (and that number will increase), we chose to make the connection dependencies optional. We could exempt postgres, but I don't think it's necessary, when you can use `pip install 'data-diff[postgresql]'` for it.

**Comment by erezsh** [comment]
Please run the following and paste the output:

```
python -m pip install data-diff[postgresql]
python -c "import psycopg2"
```

**Comment by kning** [comment]
```
(env) ➜  analysis python -m pip install "data-diff[postgresql]"

Requirement already satisfied: data-diff[postgresql] in ./env/lib/python3.10/site-packages (0.2.4)
Requirement already satisfied: click<9.0,>=8.1 in ./env/lib/python3.10/site-packages (from data-diff[postgresql]) (8.1.3)
Requirement already satisfied: toml<0.11.0,>=0.10.2 in ./env/lib/python3.10/site-packages (from data-diff[postgresql]) (0.10.2)
Requirement already satisfied: runtype<0.3.0,>=0.2.6 in ./env/lib/python3.10/site-packages (from data-diff[postgresql]) (0.2.7)
Requirement already satisfied: dsnparse in ./env/lib/python3.10/site-packages (from data-diff[postgresql]) (0.1.15)
Requirement already satisfied: rich<11.0.0,>=10.16.2 in ./env/lib/python3.10/site-packages (from data-diff[postgresql]) (10.16.2)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in ./env/lib/python3.10/site-packages (from rich<11.0.0,>=10.16.2->data-diff[postgresql]) (0.9.1)
Requirement already satisfied: colorama<0.5.0,>=0.4.0 in ./env/lib/python3.10/site-packages (from rich<11.0.0,>=10.16.2->data-diff[postgresql]) (0.4.5)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in ./env/lib/python3.10/site-packages (from rich<11.0.0,>=10.16.2->data-diff[postgresql]) (2.12.0)
WARNING: You are using pip version 22.0.4; however, version 22.2 is available.
You should consider upgrading via the '/Users/findigs/code/analysis/env/bin/python -m pip install --upgrade pip' command.
(env) ➜  analysis python -c "import psycopg2"
```

Note that I did end up manually just pip installing psycopg2 and it worked fine. Just noting that it's not ideal to have to install additional dependencies manually.

**Comment by erezsh** [comment]
I agree with you. I'm just surprised it didn't work, and I was guessing that perhaps your `pip` and `python -m pip` are not the same python installation.

Here's what I'm getting:

```
> pip install "data-diff[postgresql]"
Requirement already satisfied: data-diff[postgresql] in c:\python310\lib\site-packages (0.2.1)
Requirement already satisfied: rich<11.0.0,>=10.16.2 in c:\python310\lib\site-packages (from data-diff[postgresql]) (10.16.2)
Requirement already satisfied: dsnparse in c:\users\erez\appdata\roaming\python\python310\site-packages (from data-diff[postgresql]) (0.1.15)
Requirement already satisfied: click<9.0,>=8.1 in c:\python310\lib\site-packages (from data-diff[postgresql]) (8.1.3)
Requirement already satisfied: runtype<0.3.0,>=0.2.6 in c:\users\erez\appdata\roaming\python\python310\site-packages (from data-diff[postgresql]) (0.2.6)
Requirement already satisfied: psycopg2 in c:\python310\lib\site-packages (from data-diff[postgresql]) (2.9.3)
Requirement already satisfied: colorama in c:\users\erez\appdata\roaming\python\python310\site-packages (from click<9.0,>=8.1->data-diff[postgresql]) (0.4.5)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in c:\python310\lib\site-packages (from rich<11.0.0,>=10.16.2->data-diff[postgresql]) (2.10.0)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in c:\users\erez\appdata\roaming\python\python310\site-packages (from rich<11.0.0,>=10.16.2->data-diff[postgresql]) (0.9.1)
```

**Comment by erezsh** [comment]
Solved in #232

**Comment by erezsh** [comment]
(Next release is planned for this Monday)
